lib/SPIRV/SPIRVReader.cpp

```diff
@@ -3692,8 +3692,25 @@ void SPIRVToLLVM::transIntelFPGADecorations(SPIRVValue *BV, Value *V) {
       Value *BaseInst =
           AL ? Builder.CreateBitCast(V, Int8PtrTyPrivate, V->getName()) : Inst;
 
+      // Try to find alloca instruction for statically allocated variables.
+      // Alloca might be hidden by a couple of casts.
+      bool isStaticMemoryAttribute = AL ? true : false;
+      if (!isStaticMemoryAttribute) {
+        do {
+          if (!isa<BitCastInst>(Inst) && !isa<AddrSpaceCastInst>(Inst))
+            break;
+          Inst = dyn_cast<Instruction>(Inst->getOperand(0));
+          if (!Inst)
+            break;
+          if (isa<AllocaInst>(Inst)) {
+            isStaticMemoryAttribute = true;
+            break;
+          }
+        } while (Inst);
```

Comment:
I assume you were looking for [`Value::stripPointerCasts`](https://llvm.org/doxygen/classllvm_1_1Value.html#a966eb231e7d4e572874d2cb49b18faea)?

Reply:
Looks like it won't work. It "Strip off pointer casts, **all-zero GEPs** and address space casts", which breaks the current behavior. If we annotates structure members (to which we have access through GEP), the annotation shall be `ptr.annotation` and if we use a result of `stripPointerCasts` here, `ptr.annotation` would change to `var.annotation`.

Reply:
Or actually, we can ask alloca what type is has.

Reply:
Nevermind. The problem here is that for annotated struct objects we shall use var.annotation and for struct members we shall use ptr.annotation. And the cleanest way to distinguish them in LLVM IR is by looking on existence of GEP instruction (asking for a type of alloca won't help, since in both cases it would be a pointer on a structure). So in the end the existing approach can't be replaced by `stripPointerCasts`.